### DISCUSSION TITLE
Flutter vs Thymeleaf

### DISCUSSION THREAD

**Discussion by sangwonsheep** [Kernel360/E2E2-CATCHLINE#49] [General]
현재 `@RestController`로만 개발을 경험해보셔서 오히려 Thymeleaf 사용에 어려움이 있으신 것 같습니다.
그래서 고민이 되는 것이 서버 사이드 렌더링 vs 클라이언트 사이드 렌더링입니다.

1. Thymeleaf : 서버에서 모두 개발하기 때문에 경험해보는 것도 중요하다.
단, api를 rest하게 작성하기 어려워질 수 있다. 화면까지 api를 정해야 하기 때문에 문제가 될 수 있다.
`@ModelAttrubite`로 인해 Setter가 강제되는 문제도 있을 수 있다.
2. Flutter : Dart 언어 기반의 클라이언트 사이드 렌더링이기 때문에 rest api로 개발 가능하다.
1~2일이면 다 배울 수 있다. 문법이 많지 않기 때문에 Thymeleaf와 같이 쉽게 접근이 가능하다.
웹보다 앱에 적합한 언어이기 때문에 현재 프로젝트가 앱으로 간다면 Flutter도 괜찮은 선택지이다.

팀원분들 모두 고민해보시고, 더 좋은 방향으로 가면 좋을 것 같습니다.
Thymeleaf를 배우는 시간이나 Flutter를 배우는 시간 모두 비슷할 것이라 생각합니다.
추가적으로 멘토님께 여쭤보고 결정하면 좋을 것 같습니다!
Flutter 도움이 필요하면 옆 팀에게 도움을 요청해보겠습니다!

**Comment by soeunnPark** [comment]
둘 다 경험해보면 좋지만, Flutter 사용 좋은 것 같습니다!
타임리프 사용 시 그동안 시간을 들여 고민하고 결정했던 사항들을 번복해야 할 수도 있을 것 같아 아쉽습니다.
또 지금은 예외 핸들러에서 ResponseEntity로 감싸서 에러를 처리하고 있는데, 타임리프 사용 시 예외 처리 방법도 살짝 다른 것 같습니다.
개안적으로는, 강의를 통해 학습 중인데 모두 Json 객체를 주고받기 때문에 적용이 어려운 점이 있습니다. 
내일 이야기 나누고 결정하면 좋을 것 같습니다 😄 😸

**Comment by soeunnPark** [comment]
다만 Flutter 사용 시 로그인 방식을 세션 -> jwt 토큰으로 변경해야 할 것 같습니다.
